sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -1627,3 +1627,149 @@ def sdc_pandas_dataframe_groupby_impl(self, by=None, axis=0, level=None, as_inde
         return init_dataframe_groupby(self, column_id, grouped, sort)
 
     return sdc_pandas_dataframe_groupby_impl
+
+
+def sdc_pandas_dataframe_reset_index_codegen(df_type, all_params, columns, result_name, func_line):
+    """
+    Example of generated implementation:
+        def _df_reset_index_impl(self, level=None, drop=False, inplace=False, col_level=0, col_fill=""):
+          length = len(get_dataframe_data(self, 0))
```

Comment:
Will this work with empty DF? Probably it's better to use df_length_codelines function, if length is really needed.

Reply:
Eventually, will new implementation work with empty DF?

Reply:
@Rubtsowa I checked with:
df = pd.DataFrame({})
df = pd.DataFrame({}, index=[5, 8, 4, 6])
new implementation is called, but results don't match to reference results. Moreover 2nd case segfaults. For reset_index we need two more tests:
1. test_df_reset_index_drop_true_index_int, that will test drop=True
2. test_df_reset_index_empty_df (will be skipped)